lib/http.js

```diff
@@ -116,6 +117,36 @@ function get(url, options, cb) {
   return req;
 }
 
+/**
+ * Load WebSocket class from undici
+ * @returns {void}
+ */
+function loadWebSocket() {
+  if (!WebSocket) {
+    ({ WebSocket } = require('internal/deps/undici/undici'));
+  }
+}
+
+/**
+ * Load MessageEvent class from undici
+ * @returns {void}
+ */
+function loadMessageEvent() {
+  if (!MessageEvent) {
+    ({ MessageEvent } = require('internal/deps/undici/undici'));
+  }
+}
+
+/**
+ * Load CloseEvent class from undici
+ * @returns {void}
+ */
+function loadCloseEvent() {
+  if (!CloseEvent) {
+    ({ CloseEvent } = require('internal/deps/undici/undici'));
+  }
+}
+
```

Comment:
I'm trying to get a sense of the codebase and the patterns followed because although I'm a consumer for 15 years, this is a first contribution.

I followed the pattern of 'globalAgent'. I could have a check in a single function if either class was never used without the others. Is that the case?